include/dlaf/multiplication/triangular/impl.h

```diff
@@ -0,0 +1,399 @@
+                op, alpha, std::move(a_tile), std::move(b_tile), T(1.0), std::move(c_tile));
+}
+}
+
+namespace triangular_run {
+template <class Executor, class T, Device D>
+void trmm_B_panel_tile(Executor&& ex, blas::Diag diag, T alpha,
+                       hpx::shared_future<matrix::Tile<const T, D>> in_tile,
+                       hpx::future<matrix::Tile<T, D>> out_tile) {
+  hpx::dataflow(std::forward<Executor>(ex), matrix::unwrapExtendTiles(tile::trmm_o), blas::Side::Right,
+                blas::Uplo::Upper, blas::Op::NoTrans, diag, alpha, std::move(in_tile),
+                std::move(out_tile));
+}
+
+template <class Executor, class T, Device D>
+void gemm_trailing_matrix_tile(Executor&& ex, T alpha,
+                               hpx::shared_future<matrix::Tile<const T, D>> a_tile,
+                               hpx::shared_future<matrix::Tile<const T, D>> b_tile,
+                               hpx::future<matrix::Tile<T, D>> c_tile) {
+  hpx::dataflow(std::forward<Executor>(ex), matrix::unwrapExtendTiles(tile::gemm_o), blas::Op::NoTrans,
+                blas::Op::NoTrans, alpha, std::move(a_tile), std::move(b_tile), T(1.0),
+                std::move(c_tile));
+}
+}
+
+namespace triangular_rut {
+template <class Executor, class T, Device D>
+void trmm_B_panel_tile(Executor&& ex, blas::Op op, blas::Diag diag, T alpha,
+                       hpx::shared_future<matrix::Tile<const T, D>> in_tile,
+                       hpx::future<matrix::Tile<T, D>> out_tile) {
+  hpx::dataflow(std::forward<Executor>(ex), matrix::unwrapExtendTiles(tile::trmm_o), blas::Side::Right,
+                blas::Uplo::Upper, op, diag, alpha, std::move(in_tile), std::move(out_tile));
+}
+
+template <class Executor, class T, Device D>
+void gemm_trailing_matrix_tile(Executor&& ex, blas::Op op, T alpha,
+                               hpx::shared_future<matrix::Tile<const T, D>> a_tile,
+                               hpx::shared_future<matrix::Tile<const T, D>> b_tile,
+                               hpx::future<matrix::Tile<T, D>> c_tile) {
+  hpx::dataflow(std::forward<Executor>(ex), matrix::unwrapExtendTiles(tile::gemm_o), blas::Op::NoTrans,
+                op, alpha, std::move(a_tile), std::move(b_tile), T(1.0), std::move(c_tile));
+}
+}
+
+template <Backend backend, Device device, class T>
+void Triangular<backend, device, T>::call_LLN(blas::Diag diag, T alpha, Matrix<const T, device>& mat_a,
+                                              Matrix<T, device>& mat_b) {
+  using namespace triangular_lln;
+  auto executor_hp = dlaf::getHpExecutor<backend>();
+  auto executor_np = dlaf::getNpExecutor<backend>();
+  SizeType m = mat_b.nrTiles().rows();
+  SizeType n = mat_b.nrTiles().cols();
+
+  for (SizeType k = m - 1; k >= 0; --k) {
+    for (SizeType j = 0; j < n; ++j) {
+      auto kj = LocalTileIndex{k, j};
+
+      for (SizeType i = k + 1; i < m; ++i) {
+        auto& trailing_executor = (i == k + 1) ? executor_hp : executor_np;
```

Comment:
The TRMM DAG doesn't need any gemm to be HP. You can schedule them all as NP tasks. (It applies for all 8 cases).